### PR TITLE
feat(replacement): consolidate replacement rules

### PR DESCRIPTION
**Pull request by setchy**
## Changes

Consolidate replacement rule logic so it's no longer applied in multiple places throughout `lookup/index.ts`

## Context

Alternate PR to https://github.com/renovatebot/renovate/pull/21151

Resolves https://github.com/renovatebot/renovate/discussions/21133

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/workers/repository/process/lookup/index.ts [hunk 1]
```diff
@@ -327,21 +323,20 @@ export async function lookupUpdates(
...
-      logger.debug(
-        `Handle name-only replacement for ${packageName} without current version`
-      );
-
-      addReplacementUpdateIfValid(res.updates, config);
     } else {
       res.skipReason = 'invalid-value';
     }
 
+    if (isReplacementRulesConfigured(config)) {
```
Comment (viceice) [inline lib/workers/repository/process/lookup/index.ts:336]: this now has maybe the skip reason set from above, so it will probably fully ignored later

**Review by viceice — COMMENTED** [review]
otherwise LGTM

File: lib/workers/repository/process/lookup/index.ts (hunk 1 above)
Comment (setchy) [reply lib/workers/repository/process/lookup/index.ts:336]: hmmm, good callout.  perhaps https://github.com/renovatebot/renovate/pull/21151 is "safer"

Comment (setchy) [reply lib/workers/repository/process/lookup/index.ts:336]: actually, i think this makes it "ok" 

https://github.com/renovatebot/renovate/blob/7f9874c8998e10c523fff1234c66ae4e6c327ab3/lib/workers/repository/process/lookup/index.ts#L424-L426

**Review by viceice — APPROVED** [review]

**Review by secustor — APPROVED** [review]

**Pull request merged**